subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -498,8 +513,32 @@ static bool broadcast_source_found(struct bt_data *data, void *user_data)
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
 	uint32_t broadcast_id;
+	char broadcast_name[MAX_BROADCASTNAME_LEN + 1] = { 0 };
+	size_t broadcast_name_length = 0;
+	bool found_matched_broadcast_id = false;
+	bool found_matched_broadcast_name = false;
 	int err;
 
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
+
+	/* Check for matched broadcast name (in case of add_by_broadcast_name) */
+	if (data->type == BT_DATA_BROADCAST_NAME) {
+		if (broadcast_name_scan.broadcast_name[0] != '\0') {
+			broadcast_name_length = (data->data_len < sizeof(broadcast_name)) ?
+				data->data_len : sizeof(broadcast_name) - 1;
+			memcpy(broadcast_name, data->data, broadcast_name_length);
+			if (!strncmp(broadcast_name, broadcast_name_scan.broadcast_name,
+					broadcast_name_length)) {
```

Comment:
In the BT shell for the `connect-name` command we do something similar (https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/shell/bt.c#L216-L220), but where the `name` provided just needs to a substring, and not the whole name, of the remote device 

I suggest we do the same here, i.e. 

`add_by_broadcast_name broadcaster` should work for a remote device with broadcast name `audio_broadcaster_1`

Reply:
Ok, but then, for instance, if there are `audio_broadcast_1` and `audio_broadcast_2` available as broadcast names, one of them will be added randomly. Is it acceptable?

Reply:
Also, in case of using the `is_substring` function, it is currently defined locally in `bt.c`. So, it should be moved to a proper location (e.g., `/lib/utils/` or `/lib/libc/minimal/source/string/`) and defined globally. Hence, I can either move and use it, or I can temporarily copy/paste it into `bap_broadcast_assistant.c` for now, then a new issue to unify this function in the appropriate place can be created for the future.

Reply:
Indeed. I think that it makes sense that all name-filters/checks in the BT shell act the same and use `is_substring`